.github/workflows/ci.yml

```diff
@@ -5,26 +5,44 @@ on: [push, pull_request]
 
 jobs:
   test:
+    name: Run workflow tests
     runs-on: ubuntu-latest
     strategy:
       matrix:
         nxf_ver: ['19.10.0', '']
         aligner: ["--aligner 'hisat2'", "--aligner 'star'", "--pseudo_aligner 'salmon'"]
         options: ['--skipQC', '--remove_rRNA', '--saveUnaligned', '--skipTrimming', '--star_index false']
     steps:
-      - uses: actions/checkout@v2
+      - name: Check out pipeline code
+        uses: actions/checkout@v2
+
+      - name: Check if Docker / Conda changed
+        uses: technote-space/get-diff-action@v1
+        with:
+          PREFIX_FILTER: |
+            Dockerfile
+            environment.yml
+
+      - name: Build new docker image
+        if: env.GIT_DIFF
+        run: docker build . -t nfcore/rnaseq:dev
+
+      - name: Pull docker image
+        if: ${{ !env.GIT_DIFF }}
+        run: |
+          docker pull nfcore/rnaseq:dev
+          docker tag nfcore/rnaseq:dev nfcore/rnaseq:dev
```

Comment:
Can we finally get rid of this pull / tag?

Reply:
No - currently this is used if the environment doesn't change, same behaviour as before. See description about it taking 1 minute instead of 8 minutes etc...

Reply:
I meant can we have just a pull instead of a pull / tag

If the environment doesn't change, then we're pulling dev.

Except for a release, where we're building a new container anyway

Reply:
Whether this will work or not depends on which approach we take. See https://github.com/nf-core/rnaseq/pull/423#issuecomment-637569636

* If we take approach (1) then we can drop the `docker pull` completely.
* If we take approach (2) then we have much more complex tagging and still need this line to make the pipeline code match the environment.

So, no we can't remove it as it stands I think 😄

Reply:
We might want to make a separate issue out of this @MaxUlysse and then proceed / try to find out if we can get rid of this then ...?